feature/experimental/bgp/ate_tests/bgp_long_lived_graceful_restart/bgp_long_lived_graceful_restart_test.go

```diff
@@ -0,0 +1,1339 @@
+		if proc.GetName() == pName {
+			pID = proc.GetPid()
+			t.Logf("Pid of daemon '%s' is '%d'", pName, pID)
+		}
+	}
+	return pID
+}
+
+// gNOIKillProcess kills a daemon on the DUT, given its name and pid.
+func gNOIKillProcess(t *testing.T, dut *ondatra.DUTDevice, pName string, pID uint32) {
+	gnoiClient := dut.RawAPIs().GNOI().Default(t)
+	killRequest := &gnps.KillProcessRequest{Name: pName, Pid: pID, Signal: gnps.KillProcessRequest_SIGNAL_TERM, Restart: true}
+	killResponse, err := gnoiClient.System().KillProcess(context.Background(), killRequest)
+	t.Logf("Got kill process response: %v\n\n", killResponse)
+	if err != nil {
+		t.Fatalf("Failed to execute gNOI Kill Process, error received: %v", err)
+	}
+}
+
+// setBgpPolicy is used to configure routing policy on DUT.
+func setBgpPolicy(t *testing.T, dut *ondatra.DUTDevice, d *oc.Root) {
+	// Configure SetMED on DUT.
+	rp := d.GetOrCreateRoutingPolicy()
+	pdef5 := rp.GetOrCreatePolicyDefinition(setALLOWPolicy)
+	actions5 := pdef5.GetOrCreateStatement(aclStatement3).GetOrCreateActions()
+	actions5.GetOrCreateBgpActions().SetMed = oc.UnionUint32(bgpMED)
+	actions5.GetOrCreateBgpActions().SetLocalPref = ygot.Uint32(100)
+	gnmi.Update(t, dut, gnmi.OC().RoutingPolicy().Config(), rp)
+	//gnmi.Replace(t, dut, gnmi.OC().RoutingPolicy().Config(), rp)
+}
+
+// configureDutNewPeers configured five more BGP peers on subinterfaces.
+func configureDutNewPeers(t *testing.T, dut *ondatra.DUTDevice, nbrs []*bgpNeighbor) {
+	d := &oc.Root{}
+	dutConfPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	ni1 := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+	ni_proto := ni1.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	bgp := ni_proto.GetOrCreateBgp()
+
+	// Note: we have to define the peer group even if we aren't setting any policy because it's
+	// invalid OC for the neighbor to be part of a peer group that doesn't exist.
+	for _, nbr := range nbrs {
+		if nbr.isV4 {
+			pg1 := bgp.GetOrCreatePeerGroup(peerv4GrpName)
+			pg1.PeerAs = ygot.Uint32(nbr.as)
+			pg1.PeerGroupName = ygot.String(peerv4GrpName)
+			nv4 := bgp.GetOrCreateNeighbor(nbr.neighborip)
+			nv4.PeerGroup = ygot.String(peerv4GrpName)
+			nv4.PeerAs = ygot.Uint32(nbr.as)
+			nv4.Enabled = ygot.Bool(true)
+			af4 := nv4.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST)
+			af4.Enabled = ygot.Bool(true)
+			af6 := nv4.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV6_UNICAST)
+			af6.Enabled = ygot.Bool(false)
+		}
+	}
+	gnmi.Update(t, dut, dutConfPath.Config(), ni_proto)
+	fptest.LogQuery(t, "DUT BGP Config", dutConfPath.Config(), gnmi.GetConfig(t, dut, dutConfPath.Config()))
+}
+
+// configureATENewPeers configures five more new BGP peers on ATE.
+func configureATENewPeers(t *testing.T, ate *ondatra.ATEDevice, topo *ondatra.ATETopology, intfList []*ondatra.Interface) []*ixnet.BGP {
+
```

Comment:
delete blank line and add t.Helper() call

Reply:
updated.